source/server/filter_chain_manager_impl.cc

```diff
@@ -1,6 +1,15 @@
 #include "server/filter_chain_manager_impl.h"
 
 #include "common/common/empty_string.h"
+#include "common/common/fmt.h"
+#include "common/config/utility.h"
+#include "common/protobuf/utility.h"
+
+#include "server/configuration_impl.h"
+
+#include "extensions/filters/listener/well_known_names.h"
```

Comment:
Why are these needed? In general I would like to get rid of these headers (tracked in an issue I opened), so it would be nice to not include them here.

Reply:
There is a tricky flow that `tls_inspector` is implicitly added when filterchain matcher has the rule alpn. Not sure if it is needed in this PR. Will double check

Reply:
Removed header and bazel deps